repository/repository_test.go

```diff
@@ -0,0 +1,176 @@
+package repository
+
+import (
+	"encoding/json"
+	"encoding/xml"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"bathien/assignment/entity"
+	httpclient "bathien/assignment/http-client"
+)
+
+func TestGetPosts(t *testing.T) {
+	var (
```

Comment:
If you have time, you can watch this video https://www.youtube.com/watch?v=NwEuRO_w8HE, it will help a lot when coding with go